test/recipes/25-test_x509.t

```diff
@@ -485,6 +485,28 @@ ok(!run(app(["openssl", "x509", "-noout", "-dates", "-dateopt", "invalid_format"
 	     "-in", srctop_file("test/certs", "ca-cert.pem")])),
    "Run with invalid -dateopt format");
 
+# same tests w/ -multi (single cert)
+ok(run(app(["openssl", "x509", "-noout", "-multi", "-dates", "-dateopt", "rfc_822",
+	     "-in", srctop_file("test/certs", "ca-cert.pem")])),
+   "Run with rfc_8222 -dateopt format and -multi (single cert)");
+ok(run(app(["openssl", "x509", "-noout", "-multi", "-dates", "-dateopt", "iso_8601",
+	     "-in", srctop_file("test/certs", "ca-cert.pem")])),
+   "Run with iso_8601 -dateopt format and -multi (single cert)");
+ok(!run(app(["openssl", "x509", "-noout", "-multi", "-dates", "-dateopt", "invalid_format",
+	     "-in", srctop_file("test/certs", "ca-cert.pem")])),
+   "Run with invalid -dateopt format and -multi (single cert)");
+
+# same tests w/ -multi (multi cert)
+ok(run(app(["openssl", "x509", "-noout", "-multi", "-dates", "-dateopt", "rfc_822",
+	     "-in", srctop_file("test/certs", "goodcn2-chain.pem")])),
+   "Run with rfc_8222 -dateopt format and -multi (multi cert)");
+ok(run(app(["openssl", "x509", "-noout", "-multi", "-dates", "-dateopt", "iso_8601",
+	     "-in", srctop_file("test/certs", "goodcn2-chain.pem")])),
+   "Run with iso_8601 -dateopt format and -multi (multi cert)");
+ok(!run(app(["openssl", "x509", "-noout", "-multi", "-dates", "-dateopt", "invalid_format",
+	     "-in", srctop_file("test/certs", "goodcn2-chain.pem")])),
+   "Run with invalid -dateopt format and -multi (multi cert)");
+
```

Comment:
These tests are confusing because  `-multi` has little to do with `-dateformat`.
Better just add two test cases:
* one for which -multi has no effect, e.g., `openssl x509 -multi -in test/certs/ca-cert.pem`
* one for which it must have an effect, e.g, `openssl x509 -multi -in test/certs/goodcn2-chain.pem`
 and add to it a check that makes the test succeed only if really more then one cert get output.

Reply:
For the comparison, you can use, e.g.,
```
use File::Compare qw/compare/;

ok(!compare(srctop_file("test/certs", "leaf-chain.pem"), "chain.pem"), "-multi output contains all certs");
```

Reply:
Removed extraneous tests and removed extra (date) options from remaining tests.

Didn't see note regarding `compare` prior to using `compare_text` instead.  Hopefully sufficient.  Also just included that condition in the existing test.  Let me know if should be separate and/or if `compare` vs `compare_text` is better or more appropriate.

Reply:
IMO `compare_test()` is fine here, too, as well as combining the run with the check.